participant/tests/utils/db.server.test.ts

```diff
@@ -286,3 +288,72 @@ it("lists documents", async () => {
   expect(foundDocuments[0]).toEqual(mockument);
   expect(foundDocuments[1]).toEqual(mockumentTwo);
 });
+
+it("upserting a staff user looks up localagency and creates a staff user if there is none", async () => {
+  const mockAgency = getLocalAgency();
+  prismaMock.localAgency.findUnique.mockResolvedValue(mockAgency);
+  const mockStaffUser = getStaffUser(mockAgency.localAgencyId);
+  const upsertedStaffUser = await upsertStaffUser(
+    mockAgency.urlId,
+    mockStaffUser.staffUserId
+  );
+  expect(prismaMock.localAgency.findUnique).toHaveBeenCalledWith(
+    expect.objectContaining({
+      where: { urlId: "agency" },
+    })
+  );
+  // There is an updatedAt here.. but we'd have to do some fudging to
+  // not run into race conditions for timestamp values
+  expect(prismaMock.staffUser.upsert).toHaveBeenCalledWith(
+    expect.objectContaining({
+      where: { staffUserId: mockStaffUser.staffUserId },
+      create: {
+        staffUserId: mockStaffUser.staffUserId,
+        localAgencyId: mockAgency.localAgencyId,
+      },
+    })
+  );
+});
+
+it("upserting a staff user throws an exception if there is no agency", async () => {
+  const staffUserId = uuidv4();
+  prismaMock.localAgency.findUnique.mockResolvedValue(null);
+  await expect(async () => {
+    await upsertStaffUser("bogus", staffUserId);
+  }).rejects.toThrow(`Unable to find agency for bogus`);
+});
+
+it("upserting a staff user updates existing staff users", async () => {
+  const mockAgencyOne = getLocalAgency("one");
+  const mockAgencyTwo = getLocalAgency("two");
+  prismaMock.localAgency.findUnique.mockResolvedValue(mockAgencyTwo);
+  const mockStaffUser = getStaffUser(mockAgencyOne.localAgencyId);
+  prismaMock.staffUser.findUnique.mockResolvedValue(mockStaffUser);
```

Comment:
This is the wrong mock! Should be `prismaMock.staffUser.upsert.mockResolvedValue`